gui/brushmanager.py

```diff
@@ -182,6 +182,14 @@ def __init__(self, stock_brushpath, user_brushpath, app):
         # Update the history at the end of each definite input stroke.
         self.app.doc.input_stroke_ended += self._input_stroke_ended_cb
 
+        # In an ideal world, we would initialize the brush manager first and
+        # document second. This would allow us to place this line in the
+        # document's constructor like most callback initialization.
+        # Unfortunately, the brush manager constructor also depends on the
+        # document. (Because of the line above.)
+        # So I was forced to put this here.
+        self.brush_selected += self.app.doc._brush_selected_cb
+
```

Comment:
Here I think I'd prefer the `BrushManager` to call its own `_brush_selected_cb()`. Observing an event on the current object with a instance method of the current object seems cleaner.

(Observing far-off objects is fine too. But making far-off objects observe us feels weird.)